tests/unit/CoreImageTest.php

```diff
@@ -41,98 +42,434 @@ public function tearDown(): void {
+			<!-- /wp:image -->';
+
+		$query = $this->query();
+
+		// Update the post content with the block content.
+		wp_update_post(
+			[
+				'ID'           => $this->post_id,
+				'post_content' => $block_content,
+			]
+		);
+
+		$variables = [
+			'id' => $this->post_id,
+		];
+
+		$actual = graphql( compact( 'query', 'variables' ) );
+
+		$this->assertArrayNotHasKey( 'errors', $actual, 'There should not be any errors' );
+		$this->assertArrayHasKey( 'data', $actual, 'The data key should be present' );
+		$this->assertArrayHasKey( 'post', $actual['data'], 'The post key should be present' );
+
+		$block = $actual['data']['post']['editorBlocks'][0];
+
+		// WordPress 6.4+ adds layout styles, so `cssClassName` needs to be checked separately.
+		$this->assertStringContainsString( 'wp-block-image', $block['attributes']['cssClassName'] );
+		$this->assertStringContainsString( 'size-full', $block['attributes']['cssClassName'] );
+		$this->assertStringContainsString( 'is-resized', $block['attributes']['cssClassName'] );
+		unset( $block['attributes']['cssClassName'] );
+
+		$this->assertEquals( // Previously untested.
 			[
 				'width'           => '500',
 				'height'          => 500.0,
-				'alt'             => '',
+				'alt'             => 'alt-text',
 				'id'              => $this->attachment_id,
 				'src'             => 'http://mysite.local/wp-content/uploads/2023/05/online-programming-course-hero-section-bg.svg',
-				'style'           => null,
+				'style'           => wp_json_encode(
+					[
+						'color' => [
+							'duotone' => 'var:preset|duotone|purple-green',
+						],
+					]
+				),
 				'sizeSlug'        => 'full',
-				'linkClass'       => null,
-				'linkTarget'      => null,
+				'linkClass'       => 'test-link-css-class',
+				'linkTarget'      => '_blank',
 				'linkDestination' => 'none',
-				'align'           => null,
-				'caption'         => '',
-				'cssClassName'    => 'wp-block-image size-full is-resized',
+				'align'           => 'left',
+				'caption'         => 'Align left',
+				'className'       => 'test-css-class-name',
+				'url'             => 'http://mysite.local/wp-content/uploads/2023/05/online-programming-course-hero-section-bg.svg',
+				'borderColor'     => 'vivid-red',
+				'title'           => 'test-title',
+				'lock'            => wp_json_encode(
+					[
+						'move'   => true,
+						'remove' => true,
+					]
+				),
+				'anchor'          => 'test-anchor',
+
+				'rel'             => 'https://www.youtube.com/ noreferrer noopener',
+				'href'            => 'http://decoupled.local/dcf-1-0/',
+
 			],
-			$node['editorBlocks'][0]['attributes']
+			$block['attributes']
+		);
+	}
+
+	/**
+	 * Test that the CoreImage block previously untested attributes are retrieved correctly.
+	 *
+	 * Covers the following attributes:
+	 * - aspectRatio
+	 * - scale
+	 * - lightbox
+	 */
+	public function test_retrieve_core_untested_attributes(): void {
```

Comment:
Break this into two separate tests please.
At the top of each, gate it and `$this->markTestSkipped()` if its not the right WordPress version (like its done elsewhere), instead of wrapping the tests inside conditional logic.

Reply:
On it 🫡

Reply:
Done ✅